lib/sequelize.js

```diff
@@ -540,8 +541,19 @@ class Sequelize {
     }).then(connection => {
       const query = new this.dialect.Query(connection, this, options);
       const retryOptions = _.extend(this.options.retry, options.retry || {});
+      const queryPromises = [retry(() => query.run(sql, bindParameters), retryOptions)];
 
-      return retry(() => query.run(sql, bindParameters), retryOptions)
+      if (this.options.queryTimeout) {
```

Comment:
This is already possible with "undocumented" `retry.timeout` https://github.com/mickhansen/retry-as-promised

Actually just setting `retry` config to something like this should be able to replicate what this PR is trying to achieve
```js
{
  max: 1,
  timeout: 2000,
  name: 'Query'
}
```

Reply:
Really it works! Thanks for the advice

Reply:
This should definitely be documented!